cmd/brew-services.rb

```diff
@@ -417,18 +440,19 @@ def generate_plist(data = nil)
     data = data.to_s.gsub(/\{\{([a-z][a-z0-9_]*)\}\}/i) { |m| formula.send($1).to_s if formula.respond_to?($1) }.
               gsub(%r{(<key>Label</key>\s*<string>)[^<]*(</string>)}, '\1' + label + '\2')
 
-    # Force fix UserName
-    if !ServicesCli.root?
-      if data =~ %r{<key>UserName</key>}
-        # Replace existing UserName value with current user
-        data = data.gsub(%r{(<key>UserName</key>\s*<string>)[^<]*(</string>)}, '\1' + ServicesCli.user + '\2')
+    # Override plist UserName
+    if !set_user.nil?
+      if !ServicesCli.root?
+        opoo "--set-user has no effect for LaunchAgents"
```

Comment:
Other programs fail with an error for unknown options, so I'm not opposed to aligning with that convention.
I suppose if we assume that if someone is using the option they know that it needs sudo and just forgot, then an error will save them having to stop the service before running start again with sudo.